src/spooq2/transformer/enum_cleaner.py

```diff
@@ -0,0 +1,113 @@
+import sys
+import pyspark.sql.functions as F, types as T
+from pyspark.sql.column import Column
+
+from .transformer import Transformer
+
+
+class EnumCleaner(Transformer):
+    """
+    Cleanses a dataframe based on lists of allowed|disallowed values.
+
+    Example
+    -------
+    >>> transformer = EnumCleaner(
+    >>>     cleaning_definitions={
+    >>>         "status": {
+    >>>             "elements": ["active", "inactive"],
+    >>>         },
+    >>>         "version": {
+    >>>             "elements": ["", "None", "none", "null", "NULL"],
+    >>>             "mode": "disallow",
+    >>>             "default": None
+    >>>         },
+    >>>     }
+    >>> )
+
+    Parameters
+    ----------
+    cleaning_definitions : :py:class:`dict`
+        Dictionary containing column names and respective cleansing rules
+
+
+    Note
+    ----
+    Following cleansing rule attributes per column are supported:
+
+        * elements, mandatory - :class:`list`
+            A list of elements which will be used to allow or reject (based on mode) values from the input DataFrame.
+        * mode, allow|disallow, defaults to 'allow' - :any:`str`
+            "allow" will set all values which are not in the list (including None) to the (provided) default value.
+            "disallow" will set all values which are not in the list (excluding! None) to the (provided) default value.
```

Comment:
Shouldn't it be: will set all values which ARE in the list (excluding! None) to the (provided) default value